src/08_ecosystem/00_chapter.md

```diff
@@ -0,0 +1,51 @@
+Runtimes usually bundle together a *reactor* with one or more *executors*.
+Reactors can be thought of as event queues.
+When an event occurs, such as readable data becoming available in a socket,
+the reactor is responsible for adding a task to the queue to process this event.
+An executor handles execution of any tasks in the event queue.
+It keeps track of running and suspended tasks, polls futures to completion, and wakes tasks when they can make progress.
+The word "executor" is frequently used interchangeably with "runtime".
+
+## Choosing a Runtime
+There is no asynchronous runtime in the standard library, and none are officially recommended.
+The following crates provide popular runtimes.
+
+- [futures](https://docs.rs/futures/): The `futures` crate is not an async ecosystem like other crates in this list,
+but it provides useful traits like `Stream`, `Sink`, `AsyncRead`, and `AsyncWrite`, and utilities such as combinators.
+These utilities and traits may eventually become part of the standard library.
+`futures` has an executor, `futures-executor`, but a common choice is to use utilities from `futures` with an
+executor from another crate.
+- [Tokio](https://docs.rs/tokio/): A popular async ecosystem with HTTP, gRPC, and tracing frameworks.
+- [async-std](https://docs.rs/async-std/): A crate that provides asynchronous counterparts to standard library components.
+- [smol](https://docs.rs/smol/): A small, simplified async runtime.
+Provides the `Async` trait that can be used to wrap structs like `UnixStream` or `TcpListener`.
+- [fuchsia-async](https://fuchsia.googlesource.com/fuchsia/+/master/src/lib/fuchsia-async/): An executor for use in the Fuchsia OS.
+
+Not all async runtimes are compatible with each other.
+Notably, `Tokio` uses the `mio` reactor and defines its own versions of async I/O traits, including `AsyncRead` and `AsyncWrite`.
+It's not compatible with `async-std` and `smol`, which rely on the [`async-executor` crate](https://docs.rs/async-executor).
+The [`async_compat` crate](https://docs.rs/async_compat) can be used for compatibility between `Tokio` and other runtimes.
+
+If possible, libraries should not depend on specific runtimes.
+Ideally, only binaries should be responsible for scheduling and running tasks.
+
+## Single Threading vs Multithreading
+Async executors can be single-threaded or multi-threaded.
+For example, the `async-executor` crate has both a single-threaded `LocalExecutor` and a multi-threaded `Executor`.
+
+Tasks can either be run on the thread that created them or on a separate thread.
+Async runtimes often provide functionality for spawning tasks onto separate threads.
+Even if tasks are executed on separate threads, they should still be non-blocking.
+Tasks are usually also required to be `Send`, so they can be moved to separate threads.
+Some runtimes provide functions for spawning non-`Send` tasks onto a thread-local executor.
```

Comment:
nit: It can still be part of a multi-thread executor, only that this task can't be moved. Perhaps "...that are always executed on the same thread that spawned it"?

Reply:
rephrased but it still feels a bit awkward, lmk what you think.

Reply:
Looks good.

nit: "ensures a task is executed on the thread that spawned it"?

(Reducing the pluralism for clarity)